arbor/fvm_compartment.hpp

```diff
@@ -6,84 +6,15 @@
 #include <arbor/common_types.hpp>
```

Comment:
For this file, it is because it is a very specifically FVM-decomposition of the compartments into components suitable for the FVM discretization as performed now in `fvm_layout`.

Reply:
`segment` is prefixed only because it is quite a generic name being used in a fairly specific context. When it comes to cell type naming, we should definitely have consensus between the cell kind names and the class names; here I aligned with the name of the cell group type.

I think a good choice of names is going to require either another sort of mechanism-aware cell (quite likely, with one- or explicit-compartment models) or another sort of discretization of 3-d embedded morphologies (e.g. according to a finite difference scheme, but this is very unlikely to be implemented). Then we'll be forced to straighten all this out.